src/context/message-handling-context.jsx

```diff
@@ -5,6 +5,7 @@ const MessageHandlingContext = createContext();
 function MessageHandlingProvider({ children }) {
   const [errorMessage, setErrorMessage] = useState("");
   const [showSidenav, setShowSidenav] = useState(false);
+  const [gridView, setGridView] = useState(true);
```

Comment:
consider either having a separate context or better if prop drilling seems feasible upto a couple of levels then that would be better